src/web/rustdoc.rs

```diff
@@ -224,6 +225,19 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     req_path.insert(1, &name);
     req_path.insert(2, &version);
 
+    // if visiting the full path to the default target, remove the target from the path
+    // expects a req_path that looks like `/rustdoc/:crate/:version[/:target]/.*`
+    let crate_details = cexpect!(CrateDetails::new(&conn, &name, &version));
+    if req_path[3] == crate_details.metadata.default_target {
```

Comment:
Can we add a `&& req_path.len() > 4`? Just to be safe, I don't want this to panic.

Reply:
It works out, you can slice one past the bounds of the array without a problem: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ff716f47089c50c73b45839444f2a39b

Reply:
I added a test just to be on the safe side.